From f88d244390ab95f9fd28500d469fd37583f7efa7 Mon Sep 17 00:00:00 2001
From: OliverG96 <oliverarafo@gmail.com>
Date: Sun, 13 Oct 2013 00:11:03 +0100
Subject: [PATCH] init: Allow load of lpm.rc by parsing a cmdline argument

A usage example could be:
BOARD_LPM_BOOT_ARGUMENT_NAME := lpm_boot
BOARD_LPM_BOOT_ARGUMENT_VALUE := 1

Where lpm_boot is the argument to read, and 1 the value of it when kernel says when should enter offline charging mode

Change-Id: I413ac98138547cca62f20e3dce3964fcd30ad483

Conflicts:
	init/init.c
---
 init/Android.mk |  5 ++++-
 init/init.c     | 15 ++++++++++++++-
 2 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/init/Android.mk b/init/Android.mk
index 18230e5..016aac9 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -41,7 +41,10 @@ endif
 
 SYSTEM_CORE_INIT_DEFINES := BOARD_CHARGING_MODE_BOOTING_LPM \
     BOARD_CHARGING_CMDLINE_NAME \
-    BOARD_CHARGING_CMDLINE_VALUE
+    BOARD_CHARGING_CMDLINE_VALUE \
+    BOARD_LPM_BOOT_ARGUMENT_NAME \
+    BOARD_LPM_BOOT_ARGUMENT_VALUE
+
 
 $(foreach system_core_init_define,$(SYSTEM_CORE_INIT_DEFINES), \
   $(if $($(system_core_init_define)), \
diff --git a/init/init.c b/init/init.c
index 34dd649..b3c3e03 100755
--- a/init/init.c
+++ b/init/init.c
@@ -72,6 +72,11 @@ static int   bootchart_count;
 #define BOARD_CHARGING_CMDLINE_VALUE "true"
 #endif
 
+#ifndef BOARD_LPM_BOOT_ARGUMENT_NAME
+#define BOARD_LPM_BOOT_ARGUMENT_NAME "lpm_boot"
+#define BOARD_LPM_BOOT_ARGUMENT_VALUE "1"
+#endif
+
 static char console[32];
 static char bootmode[32];
 static char hardware[32];
@@ -103,6 +108,8 @@ static const char *ENV[32];
 
 static unsigned emmc_boot = 0;
 
+static unsigned lpm_bootmode = 0;
+
 static unsigned charging_mode = 0;
 
 static const char *expand_environment(const char *val)
@@ -713,6 +720,12 @@ static void import_kernel_nv(char *name, int for_emulator)
             emmc_boot = 1;
         }
 #endif
+    } else if (!strcmp(name,BOARD_LPM_BOOT_ARGUMENT_NAME)) {
+        if (!strcmp(value,BOARD_LPM_BOOT_ARGUMENT_VALUE)) {
+            lpm_bootmode = 1;
+        }
+    } else if (!strcmp(name,BOARD_CHARGING_CMDLINE_NAME)) {
+        strlcpy(battchg_pause, value, sizeof(battchg_pause));
     } else if (!strncmp(name, "androidboot.", 12) && name_len > 12) {
         const char *boot_prop_name = name + 12;
         char prop[PROP_NAME_MAX];
@@ -909,7 +922,7 @@ int audit_callback(void *data, security_class_t cls, char *buf, size_t len)
 static int charging_mode_booting(void)
 {
 #ifndef BOARD_CHARGING_MODE_BOOTING_LPM
-    return 0;
+    return lpm_bootmode;
 #else
     int f;
     char cmb;
-- 
1.8.1.2

